### ISSUE TITLE
Access control isn't allowing device subsets

### ISSUE THREAD

**Issue by mv-si** [rustdesk/rustdesk-server-pro#282]
Hi, 

I have an issue I currently can't solve with the possibilities access control gives me : 

- There are several users configured, which should have access rights to different type of device : 
  - Tech supports should be able to access : cashier desks, servers, office desks, ...
  - Accountants should be able to access cashier desks, servers
  - Admin should be able to access every device
  
- If I assign a strategy to cashier desks so that accountants can access it, I can't make a broader strategy for tech support to include both cashier desks and servers for example

- Groups can only be assigned to users
- Devices can only be assigned to one user

I cannot wrap my head around how I could solve this with the current way access control works. 
Please enlighten me or let's see how this can be changed to be able to have such configurations. 

Thanks

**Comment by mv-si** [comment]
I guess a solution would be to have an address book with a set permanent password for each user group, although they are a bit painful to configure because you need to specify the permanent password for each device each time you create an address book. 
Are there alternatives to this?

**Comment by rustdesk** [comment]
> * Tech supports should be able to access : cashier desks, servers, office desks, ...
> * Accountants should be able to access cashier desks, servers
> * Admin should be able to access every device

You can create cashier group, server group, office group, tech group, accountant group, admin group,

Create cross-group access rules 
- tech group -> cachier / servers / office groups
- accountant group -> cashier / servers group
- admin group -> all groups

**Comment by mv-si** [comment]
> > * Tech supports should be able to access : cashier desks, servers, office desks, ...
> > * Accountants should be able to access cashier desks, servers
> > * Admin should be able to access every device
> 
> You can create cashier group, server group, office group, tech group, accountant group, admin group,
> 
> Create cross-group access rules
> 
>     * tech group -> cachier / servers / office groups
> 
>     * accountant group -> cashier / servers group
> 
>     * admin group -> all groups

I don't understand your answer, as groups are not assignable to device but to users. 
We dont need to have an user for cachier, server and office group. 
And that'd also require to login on each device with the appropriate account, which we don't want. 

Another solution in mind?

**Comment by rustdesk** [comment]
> And that'd also require to login on each device with the appropriate account, which we don't want.

No need to to login on each device, you can just assign on web console.

**Comment by mv-si** [comment]
> > And that'd also require to login on each device with the appropriate account, which we don't want.
> 
> No need to to login on each device, you can just assign on web console. We do not have the other solution yet.

I'd like to know how please. 
Editing a device doesn't give me the option to do so : 
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/93332740/08905578-3b80-4cab-b986-f03cb2efc020)

Editing a group doesn't give me the option either : 
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/93332740/110df553-0a40-406f-9701-1aeb47aaa33d)

**Comment by rustdesk** [comment]
Edit below
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/60241597-096c-401e-ba51-ac6812d50bb1)

**Comment by rustdesk** [comment]
And we add a new way yesterday. Will be ready soon.

https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/fb7cd20e-734d-405c-9156-637df9e63159

**Comment by rustdesk** [comment]
https://github.com/rustdesk/rustdesk-server-pro/releases/tag/1.3.11

**Comment by mv-si** [comment]
> > And that'd also require to login on each device with the appropriate account, which we don't want.
> 
> No need to to login on each device, you can just assign on web console. We do not have the other solution yet.

Mh, still, your solution requires to create an user for each asset type, which isn't logical, and requires more users than needed (licencing...)

Please make it so groups can be assigned to devices. 
Thanks.

**Comment by rustdesk** [comment]
device group is added now,  https://www.reddit.com/r/rustdesk/comments/1isg7zs/selfhost_server_pro_150_is_ready/
